# Homework Introduction to Python

---



$\textbf{Ubah nama di bawah yaa!}$

- Nama: Qomaruzzaman


In [19]:
import numpy as np

# please import as much as you need

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.

Saran langkah-langkah pengerjaan
1. Convert terlebih dahulu kodi dalam jumlah barang (1 kodi = xyz buah sepatu)
2. Gunakan operasi matematika sederhana untuk menghitung profit menggunakan tiga buah variabel yang sudah ditentukan

In [20]:
# your code goes here!
kodi = 1
kuantitas = 20 * kodi
modal = 1000000
harga_jual = 80000
profit = harga_jual - (modal/kuantitas)
print(profit)

30000.0


Jadi, dari profit di atas, dapat disimpulkan bahwa jika 1 kodi yang dibeli Rp1.000.000 dan harga jual per sepatunya Rp50.000, profit yang didapat adalah sebesar Rp30.000.

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut :
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya :
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja

Saran langkah-langkah pengerjaan
1. Ikuti code dari soal, kemudian ekstrak value dari variable response menggunakan dot (.) content menjadi sebuah variabel baru (dictionary)
2. Gunakan library json, convert hasil content menggunakan json.loads
3. Print keys dari dictionary tersebut


In [21]:
# your code goes here!
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
response.content


import json
json.loads(response.content)

print(json.loads(response.content).keys())

dict_keys(['response'])


Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys 'response'.

# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)

Saran langkah-langkah pengerjaan
1. Tidak ada ketentuan nama kota (misal berdasarkan provinsi X atau pulau Y), dibebaskan untuk memilih. Buat nama-nama kota tersebut dalam sebuah list.
2. Gunakan fungsi sorted(X) untuk mengurutkan secara kecil-ke-besar (sesuai abjad), dimana X adalah contoh variabel list pada langkah (a)
3. Gunakan hasil poin (a), urutkan nama kota di dalam list berdasarkan abjad, untuk mengerjakan poin (b) dan (c).

In [22]:
# your code goes here!
list_kota = ['Aceh', 'Medan', 'Padang', 'Palembang', 'Riau', 'Jakarta', 'Bogor', 'Depok', 'Tangerang', 'Bekasi']

#1
sort_list_kota = sorted(list_kota)
print(sort_list_kota)

#2
print(sort_list_kota[:4])

#3
print(sort_list_kota[7:])

['Aceh', 'Bekasi', 'Bogor', 'Depok', 'Jakarta', 'Medan', 'Padang', 'Palembang', 'Riau', 'Tangerang']
['Aceh', 'Bekasi', 'Bogor', 'Depok']
['Palembang', 'Riau', 'Tangerang']


Jadi, dari operasi list di atas, dapat disimpulkan bahwa...

## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu:
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata
3. Trx maksimal

Saran langkah-langkah pengerjaan
1. Copy variabel total_transaksi yang ada di soal ke notebook kita (atau ketik ulang boleh)
2. Kemudian, hitung menggunakan function operasi dalam list yang telah dipelajari di kelas pada topik list. Print hasilnya setiap poin (a), (b), dan (c)

In [24]:
# your code goes here!
total_transaksi = np.array([ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ])
print(total_transaksi)
#1 trx minimum
trx_minimum = min(total_transaksi)
print(trx_minimum)

#2 Rata-rata
rata_rata = sum(total_transaksi) / len(total_transaksi)
print(rata_rata)
print(np.mean(total_transaksi))

#3
trx_maksimum = max(total_transaksi)
print(trx_maksimum)

[19 35 23 25 10 15 33 45 21 30]
10
25.6
25.6
45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa nilai minimal, max, rata-rata masing-masing adalah 10, 45, dan 25.6.

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut :
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c)

Saran langkah-langkah pengerjaan
1. HATI-HATI dalam pengerjaan, karena apabila poin sebelumnya salah, maka selanjutnya akan kebawa salah (misal (b) salah, maka poin (c) sampai (d) akan salah)
2. Untuk array 5 bilangan ganjil pertama boleh manual boleh menggunakan fungsi, dimulai dari 1
3. Ikuti perintah soal, gunakan numpy array, jangan list. Sesuaikan dengan permintaan soal pada setiap butir.



In [25]:
# your code goes here!
bil = np.array([1, 3, 5, 7, 9])
print(bil)

#1
operasi_array_1 = bil * 10
print(operasi_array_1)

#2
array_baru = np.array([-45, 43, 3, 56, 5])
print(array_baru)
operasi_array_2 = operasi_array_1 * array_baru
print(operasi_array_2)

#3
array_tertinggi = np.argmax(operasi_array_2)
array_terendah = np.argmin(operasi_array_2)
print(array_tertinggi)
print(array_terendah)

[1 3 5 7 9]
[10 30 50 70 90]
[-45  43   3  56   5]
[-450 1290  150 3920  450]
3
0


Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa array tertinggi berada pada indeks 3 dan array terendah ada pada indeks 0.

# Terimakasih, salam dari Jakarta